src/storage/v2/indices/vector_index.cpp

```diff
@@ -0,0 +1,196 @@
+      {"jaccard", unum::usearch::metric_kind_t::jaccard_k}};
+
+  auto it = metric_map.find(metric_str);
+  if (it != metric_map.end()) {
+    return it->second;
+  }
+  throw std::invalid_argument("Unknown metric kind: " + metric_str);
+}
+
+/// Helper function that converts a string representation of a scalar kind to the corresponding
+/// `unum::usearch::scalar_kind_t` value.
+unum::usearch::scalar_kind_t GetScalarKindFromConfig(const std::string &scalar_str) {
+  static const std::unordered_map<std::string, unum::usearch::scalar_kind_t> scalar_map = {
+      {"b1x8", unum::usearch::scalar_kind_t::b1x8_k}, {"u40", unum::usearch::scalar_kind_t::u40_k},
+      {"uuid", unum::usearch::scalar_kind_t::uuid_k}, {"bf16", unum::usearch::scalar_kind_t::bf16_k},
+      {"f64", unum::usearch::scalar_kind_t::f64_k},   {"f32", unum::usearch::scalar_kind_t::f32_k},
+      {"f16", unum::usearch::scalar_kind_t::f16_k},   {"f8", unum::usearch::scalar_kind_t::f8_k},
+      {"u64", unum::usearch::scalar_kind_t::u64_k},   {"u32", unum::usearch::scalar_kind_t::u32_k},
+      {"u16", unum::usearch::scalar_kind_t::u16_k},   {"u8", unum::usearch::scalar_kind_t::u8_k},
+      {"i64", unum::usearch::scalar_kind_t::i64_k},   {"i32", unum::usearch::scalar_kind_t::i32_k},
+      {"i16", unum::usearch::scalar_kind_t::i16_k},   {"i8", unum::usearch::scalar_kind_t::i8_k}};
+
+  auto it = scalar_map.find(scalar_str);
+  if (it != scalar_map.end()) {
+    return it->second;
+  }
+  throw std::invalid_argument("Unknown scalar kind: " + scalar_str);
+}
+
+// The `Impl` structure implements the underlying functionality of the `VectorIndex` class.
+// It uses the PIMPL (Pointer to Implementation) idiom to separate the interface of `VectorIndex`
+// from its implementation, making it easier to maintain, extend, and hide implementation details.
+struct VectorIndex::Impl {
+  Impl() = default;
+  ~Impl() = default;
+
+  // The `index_` member is a map that associates a `LabelPropKey` (a combination of label and property)
+  // with the actual vector index (`mg_vector_index_t`). This allows us to store multiple vector indexes
+  // based on different labels and properties.
+  std::map<LabelPropKey, mg_vector_index_t> index_;
+
+  // The `index_name_to_label_prop_` is a hash map that maps an index name (as a string) to the corresponding
+  // `LabelPropKey`. This allows the system to quickly resolve an index name to the specific label and property
+  // associated with that index, enabling easy lookup and management of indexes by name.
+  absl::flat_hash_map<std::string, LabelPropKey> index_name_to_label_prop_;
+};
+
+VectorIndex::VectorIndex() : pimpl(std::make_unique<Impl>()) {}
+VectorIndex::~VectorIndex() {}
+
+void VectorIndex::CreateIndex(const VectorIndexSpec &spec) {
+  // size and limit are important parameters for the usearch index to work properly.
+  uint64_t vector_dimension = spec.config["dimension"];
+  uint64_t limit = spec.config["limit"];
```

Comment:
What happens if the limit is reached -> document this under https://github.com/memgraph/documentation/pull/1005